### PR TITLE
Update List price Pricing link on the general tab to navigate to the Pricing tab

### LINKED ISSUE #37931: Product Block Editor: Update List price `Pricing` link on the general tab to navigate to the Pricing tab.
**Description**

The pricing link displayed in the helper text under the list price field within the general tab should not link to settings, but instead to the Pricing tab within the editor.
We can then also remove the sub-text of the list price field within the pricing section in the pricing tab.

Image

**Acceptance criteria**

- [ ] Pricing link in list price field within General tab should link to the Pricing tab within the block editor ( `&tab=pricing` )
- [ ] Remove the `Manage more settings in Pricing` sub text from the list price field within the pricing tab page.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/37931

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Got to `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
2. Under Features tab make sure to enable `product-block-editor`
3. Then visit `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. Under `General` tab and within `Basic details` section the `List price` field helper text should be `Manage more settings in Pricing`.
5. When click the `Pricing` the `Pricing` tab should be shown and the `List price` field helper text should not be shown.

### CODE REVIEW

File: packages/js/product-editor/src/blocks/pricing/edit.tsx [hunk 1]
```diff
@@ -38,51 +42,44 @@ export function Edit( { attributes }: { attributes: BlockAttributes } ) {
...
-					'Manage more settings in <link>Pricing.</link>',
-					'woocommerce'
+	const interpolatedHelp = help
+		? createInterpolateElement( help, {
+				PricingTab: (
+					<Link
+						href={ getNewPath( { tab: 'pricing' } ) }
+						onClick={ () => {
+							recordEvent(
+								'product_pricing_list_price_help_tax_settings_click'
```
Comment (louwie17) [inline packages/js/product-editor/src/blocks/pricing/edit.tsx]: Nice, thanks for adding tracks. Although not sure if this is related to the `help_tax_settings`, is that a copy/paste error maybe?
We may want to simplify it a bit to: `product_pricing_help_click`?

**Review by louwie17 — COMMENTED** [review]
Nice work @mdperez86, also appreciate the additional re-factors to clean things up 💯 
I just left one minor suggestion around a tracks event name, otherwise this looks great!

File: packages/js/product-editor/src/blocks/pricing/edit.tsx (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/blocks/pricing/edit.tsx]: Well that was not part of the scope of this PR but it was solved here anyway -> https://github.com/woocommerce/woocommerce/pull/37961/commits/db3bb644b2038bb1b9d89146c5a67316906e34b6

**Review by louwie17 — APPROVED** [review]
Thanks for updating the track action name, LGTM 🚀

**Pull request merged**